sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -210,13 +210,25 @@ case class EqualNullSafe(left: Expression, right: Expression) extends BinaryComp
 case class LessThan(left: Expression, right: Expression) extends BinaryComparison {
   override def symbol: String = "<"
 
-  lazy val ordering: Ordering[Any] = {
+  lazy val orderings: Either[Ordering[Any], (Int, Ordering[Any])] = {
```

Comment:
let's not use either. it is a super confusing data type. here i think you actually want an ordering defined for complex types.